### PR TITLE
[libass] update to 0.16.0

### PR DESCRIPTION
**Pull request by Masaiki**
rewrite part of the CMakeLists.txt for matching the lastest configure.ac

### CODE REVIEW

File: ports/libass/CMakeLists.txt [hunk 1]
```diff
@@ -45,8 +83,10 @@ find_path(HARFBUZZ_INCLUDE_DIR
           NAMES hb.h
           PATH_SUFFIXES harfbuzz)
 
-find_path(DIRENT_INCLUDE_DIR
-          NAMES dirent.h)
+if (NOT WIN32)
+  find_path(DIRENT_INCLUDE_DIR
+            NAMES dirent.h)
+endif()
```
Comment (FrankXie05) [inline ports/libass/CMakeLists.txt:89]:
```
if (NOT WIN32)
  find_path(DIRENT_INCLUDE_DIR
            NAMES dirent.h)
endif()
```
If the port `libass` is no longer required to depend on `dirent` please remove it and attach a reason.
If you just change the triplets supported  by `dirent`, please modify it in the field `dependencies` of `vcpkg.json` and add the reason.(I saw you just deleted it directly in `vcpkg.json`)

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/libass/CMakeLists.txt (hunk 1 above)
Comment (Masaiki) [reply ports/libass/CMakeLists.txt]: Done.

File: ports/libass/CMakeLists.txt [hunk 2]
```diff
@@ -1,38 +1,76 @@
...
   list(FILTER SOURCES EXCLUDE REGEX ".*ass_fontconfig.c$")
+
+  check_c_source_compiles(
+    "
+    #include <winapifamily.h>
+    #if WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)
+    #error Win32 desktop APIs are available
+    #endif
+    "
+    CHECK_WINRT_UWP
```
Comment (FrankXie05) [inline ports/libass/CMakeLists.txt:31]: I would like to ask whether this macro is used to judge the current triplet as `uwp`? 
If so, the port `libass` dependency `fribidi` does not support uwp builds,
https://github.com/microsoft/vcpkg/blob/6dbfaba0393e851fcef2a90ed33fcdac8074d241/ports/fribidi/vcpkg.json#L5
 so the following cannot determine whether to choose `gdi `or `dwrite`.
Am I understanding correctly? :)

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/libass/CMakeLists.txt (hunk 2 above)
Comment (Masaiki) [reply ports/libass/CMakeLists.txt:31]: you are right, this has not been considered.

**Review by FrankXie05 — APPROVED** [review]

**Comment by FrankXie05** [comment]
@Masaiki One last bit of nitpicking advice, are you able to modernize the port. We're using the new function now, I'll keep it open for 12 hours, and if there's no response, we'll merge it and modernize it in the next update. Reference message https://github.com/microsoft/vcpkg/pull/24749#pullrequestreview-976668175  
😀

**Comment by Masaiki** [comment]
> @Masaiki One last bit of nitpicking advice, are you able to modernize the port. We're using the new function now, I'll keep it open for 12 hours, and if there's no response, we'll merge it and modernize it in the next update. Reference message [#24749 (review)](https://github.com/microsoft/vcpkg/pull/24749#pullrequestreview-976668175) 😀

Done. Thanks for your following.

**Review by FrankXie05 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**